WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Orders/LastOrdersDashboardCardViewModelTests.swift

```diff
@@ -0,0 +1,225 @@
+            switch action {
+            case let .synchronizeOrders(_, _, _, _, _, _, _, _, _, completion):
+                XCTAssertTrue(viewModel.syncingData)
+                completion(1, nil)
+            default:
+                break
+            }
+        }
+
+        await viewModel.reloadData()
+
+        // Then
+        XCTAssertFalse(viewModel.syncingData)
+    }
+
+    @MainActor
+    func test_syncingError_is_updated_correctly_when_loading_orders_fails() async {
+        // Given
+        let viewModel = LastOrdersDashboardCardViewModel(siteID: sampleSiteID,
+                                                         stores: stores,
+                                                         storageManager: storageManager)
+        XCTAssertNil(viewModel.syncingError)
+        let error = NSError(domain: "test", code: 500)
+        mockOrderStatuses()
+
+        // When
+        stores.whenReceivingAction(ofType: OrderAction.self) { action in
+            switch action {
+            case let .synchronizeOrders(_, _, _, _, _, _, _, _, _, completion):
+                completion(1, error)
+            default:
+                break
+            }
+        }
+        await viewModel.reloadData()
+
+        // Then
+        XCTAssertEqual(viewModel.syncingError as? NSError, error)
+    }
+
+
+    @MainActor
+    func test_order_statuses_are_loaded_from_storage_when_available() async {
+        // Given
+        let viewModel = LastOrdersDashboardCardViewModel(siteID: sampleSiteID,
+                                                         stores: stores,
+                                                         storageManager: storageManager)
+        insertOrderStatuses(sampleOrderStatuses)
+
+        mockSynchronizeOrders()
+        mockOrderStatuses()
+
+        // When
+        await viewModel.reloadData()
+
+        let statusSlugs = sampleOrderStatuses
+            .map { $0.slug }
+            .sorted()
+
+        // Then
+        XCTAssertEqual(viewModel.allStatuses.map { $0.id }, (["any"] + statusSlugs))
+    }
+
+    @MainActor
+    func test_orders_are_loaded_based_on_selected_status() async {
+        // Given
+        let viewModel = LastOrdersDashboardCardViewModel(siteID: sampleSiteID,
+                                                         stores: stores,
+                                                         storageManager: storageManager)
+        insertOrders(sampleOrders)
+        mockSynchronizeOrders()
+        mockOrderStatuses()
+
+        // When
+        await viewModel.updateOrderStatus(.cancelled)
+
+        // Then
+        let orderIDs = Array(
+            sampleOrders
+                .filter({ $0.status == .cancelled })
+                .sorted(by: { $0.dateCreated > $1.dateCreated })
+                .map({ $0.orderID })
+                .prefix(3)
+        )
+        XCTAssertEqual(viewModel.rows.map({ $0.id }), orderIDs)
+    }
+}
+
+extension LastOrdersDashboardCardViewModelTests {
```

Comment:
Super nit:

```suggestion
private extension LastOrdersDashboardCardViewModelTests {
```

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/12847/commits/e35623001255d54fcc1035d77f9e090a6f5a92ef